napari/_tests/test_viewer.py

```diff
@@ -61,6 +61,16 @@ def test_no_qt_loop():
         _ = Viewer()
 
 
+def test_3d_image_pyramid(viewer_factory):
+    view, viewer = viewer_factory()
+    data = [np.random.random((128,) * 3), np.random.random((64,) * 3)]
+    viewer.add_image(data)
+    # check that this doesn't crash
+    viewer.dims.ndisplay = 3
+    viewer.layers[0].refresh()
+    check_viewer_functioning(viewer, view=view, data=data, ndim=3)
```

Comment:
This passes on master though it crashes when I try to run the viewer on my machine (and I expect all machines). Any ideas why?

Reply:
The reason is because the vispy `on_draw` command isn't triggered so the code with the bug doesn't get called in our tests see here for an example https://github.com/napari/napari/blob/master/napari/_vispy/_tests/test_vispy_multiscale.py#L18

I will move the test into that folder and add the `on_draw` command

Reply:
done